### PR TITLE
[Merged by Bors] - Add note about global `.gitignore` to `CONTRIBUTING.md` — Instead of ignoring `.DS_Store` files created by macOS Finder

### PR DESCRIPTION
**Pull request by coreh**
# Objective

Finder in macOS creates hidden `.DS_Store` files containing metadata (for icon positioning, view mode, etc) whenever you browse a directory. There's no point in committing these to git, and they're a common git + macOS nuisance.

## Solution

- ~~This PR adds `.DS_Store` files to `.gitignore`, improving the developer experience on macOS.~~
- This PR adds a note to the `CONTRIBUTING.md` file teaching how to use global git ignore.

### CODE REVIEW

**Comment by mockersf** [comment]
you should add that to your global gitignore, not to each project you'll contribute to

**Comment by alice-i-cecile** [comment]
I agree with Francois here :)

**Comment by MrGVSV** [comment]
> you should add that to your global gitignore, not to each project you'll contribute to

I disagree. Why should Bevy define a .got ignore at all then?

We can leave this out if that's what people prefer but I don't see the harm in adding it haha

**Comment by mockersf** [comment]
> Why should Bevy define a .gitignore at all then?

for files to ignore that are specific to this project and that you don't want to ignore on all projects, like a `Cargo.lock` file, or files generated during tests. the local .gitignore file is also useful to ignore files from a project that have default values but that users may want to customise without committing them

**Comment by Vrixyz** [comment]
TL;DR: I approve this addition.

Even though I tend to agree with @mockersf within a « trusted team », in an open source project I think it makes sense to add it:
Pros:
- it helps newcomers and reviewers
- Only 1 line
Cons:
- platform specific, can (and should) be enabled globally on a machine.
- Still 1 line, what if everyone starts adding their own specific edge case ignores..?

In my opinion the suggestive advantage balance is on the pros’ side 👍.

Current .gitignore also contains .vscode and .idea ; which are not machine-wide files, but independent developer experience files… in my opinion, if we want to be purist about our gitignore, we would want to set those ignores locally rather than from our project ( https://stackoverflow.com/questions/1753070/how-do-i-configure-git-to-ignore-some-files-locally )

But because a lot of people are using vscode or itellij/clion, it’s worth it to add them 👍 I would argue .DSStore is comparable to that :)

**Comment by mockersf** [comment]
> But because a lot of people are using vscode or itellij/clion, it’s worth it to add them 

You're right, we should remove them 🥷 

GitHub as a reference repo for .gitignore files. macOS specific files and editors are in the Global folder 😉  https://github.com/github/gitignore/tree/main/Global

I would prefer to add doc to the `CONTRIBUTING.md`, maybe just a link to https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

**Comment by MrGVSV** [comment]
> If that's really an issue for people, I would prefer to add doc to the `CONTRIBUTING.md`, maybe just a link to https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

I think this would be a nice compromise if we decide to not include this change. I didn't even know you could ignore files in git globally 😅

**Comment by coreh** [comment]
Pivoted the PR purpose 😅 It now adds a `CONTRIBUTING.md` note. That tip was personally helpful to me, since I don't remember being aware of or ever using global `.gitignore` before.

I believe that's a little less controversial, but if y'all want I can un-pivot the purpose of the PR, I don't have a strong preference either way now.

**Review by alice-i-cecile — APPROVED** [review]

**Review by MrGVSV — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Comment by coreh** [comment]
Looks like this fail to build due to an issue in CI?

```
Cannot add PPA: 'ppa:~oibaf/ubuntu/graphics-drivers'.
[11](https://github.com/bevyengine/bevy/actions/runs/3565094148/jobs/5989837932#step:5:12)
The user named '~oibaf' has no PPA named 'ubuntu/graphics-drivers'
[12](https://github.com/bevyengine/bevy/actions/runs/3565094148/jobs/5989837932#step:5:13)
Please choose from the following available PPAs:
[13](https://github.com/bevyengine/bevy/actions/runs/3565094148/jobs/5989837932#step:5:14)
 * 'gallium-nine':  Gallium Nine Test PPA
[14](https://github.com/bevyengine/bevy/actions/runs/3565094148/jobs/5989837932#step:5:15)
 * 'graphics-drivers':  Updated Open Graphics Drivers - since 2011!
[15](https://github.com/bevyengine/bevy/actions/runs/3565094148/jobs/5989837932#step:5:16)
 * 'test':  Testing PPA
```

**Comment by alice-i-cecile** [comment]
bors retry
